include/libopencm3/stm32/h7/rcc.h

```diff
@@ -0,0 +1,458 @@
+#define RCC_APB1HRSTR             MMIO32(RCC_BASE + 0x094)
+#define RCC_APB2RSTR              MMIO32(RCC_BASE + 0x098)
+#define RCC_APB3RSTR              MMIO32(RCC_BASE + 0x08C)
+#define RCC_APB4RSTR              MMIO32(RCC_BASE + 0x09C)
+#define RCC_AHB1ENR               MMIO32(RCC_BASE + 0x0D8)
+#define RCC_AHB2ENR               MMIO32(RCC_BASE + 0x0DC)
+#define RCC_AHB3ENR               MMIO32(RCC_BASE + 0x0D4)
+#define RCC_AHB4ENR               MMIO32(RCC_BASE + 0x0E0)
+#define RCC_APB1LENR              MMIO32(RCC_BASE + 0x0E8)
+#define RCC_APB1HENR              MMIO32(RCC_BASE + 0x0EC)
+#define RCC_APB2ENR               MMIO32(RCC_BASE + 0x0F0)
+#define RCC_APB3ENR               MMIO32(RCC_BASE + 0x0E4)
+#define RCC_APB4ENR               MMIO32(RCC_BASE + 0x0F4)
+#define RCC_AHB1LPENR             MMIO32(RCC_BASE + 0x100)
+#define RCC_AHB2LPENR             MMIO32(RCC_BASE + 0x104)
+#define RCC_AHB4LPENR             MMIO32(RCC_BASE + 0x108)
+#define RCC_APB1LLPENR            MMIO32(RCC_BASE + 0x110)
+#define RCC_APB1HLPENR            MMIO32(RCC_BASE + 0x114)
+#define RCC_APB2LPENR             MMIO32(RCC_BASE + 0x118)
+#define RCC_APB3LPENR             MMIO32(RCC_BASE + 0x10C)
+#define RCC_APB4LPENR             MMIO32(RCC_BASE + 0x11C)
+#define RCC_BDCR                  MMIO32(RCC_BASE + 0x70)
+#define RCC_CSR                   MMIO32(RCC_BASE + 0x74)
+#define RCC_SSCGR                 MMIO32(RCC_BASE + 0x80)
+#define RCC_PLLI2SCFGR            MMIO32(RCC_BASE + 0x84)
+#define RCC_PLLSAICFGR            MMIO32(RCC_BASE + 0x88)
+#define RCC_DCKCFGR1              MMIO32(RCC_BASE + 0x8C)
+#define RCC_DCKCFGR2              MMIO32(RCC_BASE + 0x90)
+
+/* --- RCC_CFGR values ----------------------------------------------------- */
+/* TODO: Confirm functionality of these blocks. */
+
+/* MCO2: Microcontroller clock output 2 */
+#define RCC_CFGR_MCO2_MASK        0x7
+#define RCC_CFGR_MCO2_SHIFT       29
+#define RCC_CFGR_MCO2_SYSCLK      0x0
+#define RCC_CFGR_MCO2_PLL2        0x1
+#define RCC_CFGR_MCO2_HSE         0x2
+#define RCC_CFGR_MCO2_PLL1        0x3
+#define RCC_CFGR_MCO2_CSI         0x4
+#define RCC_CFGR_MCO2_LSI         0x5
+
+/* MCO1/2PRE: MCO Prescalers */
+#define RCC_CFGR_MCOPRE_MASK      0xf
+#define RCC_CFGR_MCO2PRE_SHIFT    25
+#define RCC_CFGR_MCO1PRE_SHIFT    18
+#define RCC_CFGR_MCOPRE_DIV_NONE  0x0
+#define RCC_CFGR_MCOPRE_DIV_BYP   0x1
+#define RCC_CFGR_MCOPRE_DIV_2     0x2
+#define RCC_CFGR_MCOPRE_DIV_3     0x3
+#define RCC_CFGR_MCOPRE_DIV_4     0x4
+/* Note: MCOPRE_DIV can go from 1 - 15  */
+
+/* MCO1: Microcontroller clock output 1 */
+#define RCC_CFGR_MCO1_MASK        0x7
+#define RCC_CFGR_MCO1_SHIFT       22
+#define RCC_CFGR_MCO1_HSI         0x0
+#define RCC_CFGR_MCO1_LSE         0x1
+#define RCC_CFGR_MCO1_HSE         0x2
+#define RCC_CFGR_MCO1_PLL1        0x3
+#define RCC_CFGR_MCO1_HSI48       0x4
+#define RCC_CFGR_MCO_SHIFT        RCC_CFGR_MCO1_SHIFT
+#define RCC_CFGR_MCO_MASK         RCC_CFGR_MCO1_MASK
```

Comment:
what? what is this for?

Reply:
`stm32/common/rcc_common_all.c:	RCC_CFGR = (RCC_CFGR & ~(RCC_CFGR_MCO_MASK << RCC_CFGR_MCO_SHIFT)) ...`

Reply:
Well it won't work on mco2 stuff anyway.  Would it perhaps make more sense to update the common code to all be MCO1?

Reply:
Well...I think it makes sense to add an additional call for MCO2, but I don't know that it makes sense to change the common API for this one special snowflake. 

I think adding entry points for explicit MCO1 and MCO2 to the h7/rcc.h and h7/rcc.c will certainly make sense, but maybe not for the common code. This could also be solved with ifdefs in the common file.

Overall though, I didn't think it made sense to modify existing APIs _yet_ for this implementation.

Reply:
reasonable, just wanted to bring it up, see where it was going.